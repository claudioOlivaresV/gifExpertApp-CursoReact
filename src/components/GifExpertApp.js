import React, { useState } from 'react'
import { AddCategory } from './addCategory/AddCategory'
import { GiftGrid } from './giftGrid/GiftGrid'
export const GifExpertApp = () => {
    // const categories = ['One Puch', 'Samurai X', 'Dragon Ball']

    const [categories, setCategories] = useState(['Dragon Ball'])

    // const handleAdd = () => {
    //     // setCategories([...categories, 'nueva'])
    //     setCategories(category => [...category, 'nueva'])

    // }
    return (
        <>
            <h1>GifExpertApp </h1>
            <hr />
            {/* <button onClick={handleAdd}>Agregar</button> */}
            <AddCategory setCategories={setCategories}/>
            <ol>
                {
                    categories.map((category) => 
                    <GiftGrid  key={category} category={category}/>
                    )
                }
            </ol>

        </>
    )
}
