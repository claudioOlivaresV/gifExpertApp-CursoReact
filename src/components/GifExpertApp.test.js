import '@testing-library/jest-dom'
import { shallow } from 'enzyme'
import { GifExpertApp } from './GifExpertApp'
describe('Test Gif ExpertApp', () => {
    const wrapper = shallow (<GifExpertApp />)
    test('Probar snapshot', async () => {
        expect(wrapper).toMatchSnapshot();
    })
})
