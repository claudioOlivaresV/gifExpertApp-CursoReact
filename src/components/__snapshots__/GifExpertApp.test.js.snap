// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Gif ExpertApp Probar snapshot 1`] = `
<Fragment>
  <WithStyles(SimpleAppBar) />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GiftGrid
      category="Dragon Ball Z"
      key="Dragon Ball Z"
    />
  </ol>
</Fragment>
`;
